core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java

```diff
@@ -146,22 +147,28 @@ private synchronized void fetchDynamicFiltersIfNecessary()
             return;
         }
 
-        Request request = prepareGet()
-                .setUri(uriBuilderFrom(taskUri).appendPath("dynamicfilters").build())
-                .setHeader(CONTENT_TYPE, JSON_UTF_8.toString())
-                .setHeader(TRINO_CURRENT_VERSION, Long.toString(localDynamicFiltersVersion))
-                .setHeader(TRINO_MAX_WAIT, refreshMaxWait.toString())
-                .build();
-
+        Request request = prepareDynamicFiltersRequest();
         errorTracker.startRequest();
         future = httpClient.executeAsync(request, createFullJsonResponseHandler(dynamicFilterDomainsCodec));
-        addCallback(future, new SimpleHttpResponseHandler<>(new DynamicFiltersResponseCallback(), request.getUri(), stats), executor);
+        addCallback(
+                future,
+                new SimpleHttpResponseHandler<>(
+                        new DynamicFiltersResponseCallback(this::fetchDynamicFiltersIfNecessary),
+                        request.getUri(),
+                        stats),
+                executor);
     }
 
     private class DynamicFiltersResponseCallback
             implements SimpleHttpResponseCallback<VersionedDynamicFilterDomains>
     {
         private final long requestStartNanos = System.nanoTime();
+        private final Runnable triggerNextRequest;
```

Comment:
are we sleeping between requests?

Reply:
Good point, in pipelined mode sleeping was not necessary as it would make the next request only if local version is out of date. I've update the code to add a small wait after successful request to avoid `fetchFinalDynamicFiltersIfNecessary` from retrying too aggressively.